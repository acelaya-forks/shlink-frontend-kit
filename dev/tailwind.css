@import 'tailwindcss' prefix(tw) important;
@import '../src/tailwind.preset.css';
