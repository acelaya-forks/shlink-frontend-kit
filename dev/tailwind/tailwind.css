@import 'tailwindcss' prefix(tw) important;
@import '../../src/tailwind/tailwind.preset.css';
