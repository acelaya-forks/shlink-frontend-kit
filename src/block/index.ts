export * from './Message';
export * from './Result';
export * from './SimpleCard';
