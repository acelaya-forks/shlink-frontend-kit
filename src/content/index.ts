export * from './Details';
export * from './Listbox';
export * from './Table';
