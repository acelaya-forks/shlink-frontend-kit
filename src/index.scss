@import './utils/ResponsiveTable';
@import './theme/theme';
