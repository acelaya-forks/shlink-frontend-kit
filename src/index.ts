export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
