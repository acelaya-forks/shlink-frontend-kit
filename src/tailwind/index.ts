export * from './content';
export * from './feedback';
export * from './form';
export * from './helpers';
export * from './navigation';
export * from './surfaces';
export * from './utils';
export * from './types';
