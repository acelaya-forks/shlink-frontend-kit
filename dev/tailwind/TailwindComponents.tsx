import type { FC } from 'react';
import { Route, Routes } from 'react-router';
import { DetailsPage } from './content/DetailsPage';
import { TablePage } from './content/TablePage';
import { MessagePage } from './feedback/MessagePage';
import { ModalDialogPage } from './feedback/ModalDialogPage';
import { ResultPage } from './feedback/ResultPage';
import { TooltipPage } from './feedback/TooltipPage';
import { ButtonsPage } from './form/ButtonsPage';
import { InputsPage } from './form/InputsPage';
import { SearchComboboxPage } from './form/SearchComboboxPage';
import { SearchInputPage } from './form/SearchInputPage';
import { NavPillsPage } from './navigation/NavPillsPage';
import { PaginatorPage } from './navigation/PaginatorPage';
import { CardsPage } from './surfaces/CardsPage';
import { UtilsPage } from './utils/UtilsPage';

export const TailwindComponents: FC = () => {
  return (
    <Routes>
      <Route path="/" element={<h2 className="text-center">Select component - Tailwind</h2>} />
      <Route path="/form/inputs" element={<InputsPage />} />
      <Route path="/form/search-input" element={<SearchInputPage />} />
      <Route path="/form/search-combobox" element={<SearchComboboxPage />} />
      <Route path="/form/buttons" element={<ButtonsPage />} />
      <Route path="/surfaces/cards" element={<CardsPage />} />
      <Route path="/content/tables" element={<TablePage />} />
      <Route path="/content/details" element={<DetailsPage />} />
      <Route path="/navigation/paginator" element={<PaginatorPage />} />
      <Route path="/navigation/nav-pills">
        <Route path="" element={<NavPillsPage />} />
        <Route path="*" element={<NavPillsPage />} />
      </Route>
      <Route path="/feedback/dialogs" element={<ModalDialogPage />} />
      <Route path="/feedback/result" element={<ResultPage />} />
      <Route path="/feedback/message" element={<MessagePage />} />
      <Route path="/feedback/tooltip" element={<TooltipPage />} />
      <Route path="/utils" element={<UtilsPage />} />
      <Route path="*" element={<h2 className="text-center">Not found - Tailwind</h2>} />
    </Routes>
  );
};
