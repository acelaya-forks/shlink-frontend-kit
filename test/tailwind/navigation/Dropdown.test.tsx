import { screen } from '@testing-library/react';
import type { DropdownProps } from '../../../src/tailwind';
import { Dropdown,LabelledInput  } from '../../../src/tailwind';
import { checkAccessibility } from '../../__helpers__/accessibility';
import { renderWithEvents } from '../../__helpers__/setUpTest';

describe('<Dropdown />', () => {
  const setUp = (
    props: Pick<DropdownProps, 'buttonSize' | 'buttonVariant' | 'caretless' | 'buttonDisabled'> = {},
  ) => renderWithEvents(
    <div>
      <Dropdown buttonContent="Press me" {...props}>
        <Dropdown.Item>One</Dropdown.Item>
        <Dropdown.Item>Two</Dropdown.Item>
        <Dropdown.Item>Three</Dropdown.Item>
        <Dropdown.Misc>
          <LabelledInput aria-hidden type="text" label="Text input" />
        </Dropdown.Misc>
      </Dropdown>
      <button>Other button</button>
    </div>,
  );
  const setUpOpened = async () => {
    const { user, ...rest } = setUp();

    await user.click(screen.getByRole('button', { name: 'Press me' }));
    await screen.findByRole('menu');

    return { user, ...rest };
  };

  it.each([
    setUp,
    setUpOpened,
  ])('passes a11y checks', (setUpFunction) => checkAccessibility(setUpFunction()));

  it('closes menu when pressing `Escape`', async () => {
    const { user } = await setUpOpened();

    expect(screen.getByRole('menu')).toBeInTheDocument();
    await user.type(screen.getByLabelText('Text input'), '{Escape}');
    expect(screen.queryByRole('menu')).not.toBeInTheDocument();
  });

  it('closes menu when clicking away', async () => {
    const { user } = await setUpOpened();

    expect(screen.getByRole('menu')).toBeInTheDocument();
    await user.click(screen.getByRole('button', { name: 'Other button' }));
    expect(screen.queryByRole('menu')).not.toBeInTheDocument();
  });

  it('closes menu when focusing away', async () => {
    const { user } = await setUpOpened();

    expect(screen.getByRole('menu')).toBeInTheDocument();
    await user.tab(); // Tab to focus the next focusable element, which is outside the menu
    expect(screen.queryByRole('menu')).not.toBeInTheDocument();
    expect(document.activeElement).toEqual(screen.getByRole('button', { name: 'Other button' }));
  });

  it.each([
    { buttonSize: 'sm' as const },
    { buttonSize: 'md' as const },
    { buttonSize: 'lg' as const },
    { buttonVariant: 'button' as const },
    { buttonVariant: 'link' as const },
    { caretless: false },
    { caretless: true },
    { buttonDisabled: true },
  ])('renders toggle button with the right classes based on provided props', (props) => {
    setUp(props);
    expect(screen.getByRole('button', { name: 'Press me' }).className).toMatchSnapshot();
  });

  it.each([
    { props: {} },
    { props: { caretless: true } },
    { props: { caretless: false } },
  ])('renders caret only if caretless is false', ({ props }) => {
    setUp(props);

    if (!props.caretless) {
      expect(screen.getByRole('img', { hidden: true })).toBeInTheDocument();
    } else {
      expect(screen.queryByRole('img', { hidden: true })).not.toBeInTheDocument();
    }
  });
});
