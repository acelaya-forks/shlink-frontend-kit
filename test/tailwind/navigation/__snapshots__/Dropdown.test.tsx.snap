// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Dropdown /> > renders toggle button with the right classes based on provided props 1`] = `"tw:flex tw:items-center tw:gap-x-2 tw:focus-ring tw:justify-between tw:border tw:border-lm-border tw:dark:border-dm-border tw:rounded-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input tw:px-1.5 tw:py-1 tw:text-sm"`;

exports[`<Dropdown /> > renders toggle button with the right classes based on provided props 2`] = `"tw:flex tw:items-center tw:gap-x-2 tw:focus-ring tw:justify-between tw:border tw:border-lm-border tw:dark:border-dm-border tw:rounded-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input tw:px-3 tw:py-1.5"`;

exports[`<Dropdown /> > renders toggle button with the right classes based on provided props 3`] = `"tw:flex tw:items-center tw:gap-x-2 tw:focus-ring tw:justify-between tw:border tw:border-lm-border tw:dark:border-dm-border tw:rounded-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input tw:px-4 tw:py-2 tw:text-lg"`;

exports[`<Dropdown /> > renders toggle button with the right classes based on provided props 4`] = `"tw:flex tw:items-center tw:gap-x-2 tw:focus-ring tw:justify-between tw:border tw:border-lm-border tw:dark:border-dm-border tw:rounded-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input tw:px-3 tw:py-1.5"`;

exports[`<Dropdown /> > renders toggle button with the right classes based on provided props 5`] = `"tw:flex tw:items-center tw:gap-x-2 tw:focus-ring tw:justify-between tw:text-lm-brand tw:dark:text-dm-brand tw:highlight:text-lm-brand-dark tw:dark:highlight:text-dm-brand-dark tw:highlight:underline tw:px-3 tw:py-1.5"`;

exports[`<Dropdown /> > renders toggle button with the right classes based on provided props 6`] = `"tw:flex tw:items-center tw:gap-x-2 tw:focus-ring tw:justify-between tw:border tw:border-lm-border tw:dark:border-dm-border tw:rounded-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input tw:px-3 tw:py-1.5"`;

exports[`<Dropdown /> > renders toggle button with the right classes based on provided props 7`] = `"tw:flex tw:items-center tw:gap-x-2 tw:focus-ring tw:border tw:border-lm-border tw:dark:border-dm-border tw:rounded-md tw:bg-lm-primary tw:dark:bg-dm-primary tw:group-[&]/card:bg-lm-input tw:group-[&]/card:dark:bg-dm-input tw:px-3 tw:py-1.5"`;
